pynamodb/attributes.py

```diff
@@ -1258,8 +1308,8 @@ def _get_serialize_class(self, value):
 
 
 DESERIALIZE_CLASS_MAP: Dict[str, Attribute] = {
-    BINARY: BinaryAttribute(),
-    BINARY_SET: BinarySetAttribute(),
+    BINARY: BinaryDataAttribute(),
```

Comment:
This will be a problem if anyone had a `BinaryAttribute` stored in a `DynamicMapAttribute` right?

Reply:
Ah, right, I meant to go back to that, since that change looked iffy. It also depends on whether `_handle_binary_attributes` handles anything recursively and looking at it, I don't think it does... 🙀